boot/zephyr/nrf_cleanup.c

```diff
@@ -105,6 +107,21 @@ void nrf_cleanup_peripheral(void)
     for (int i = 0; i < sizeof(nrf_uarte_to_clean) / sizeof(nrf_uarte_to_clean[0]); ++i) {
         NRF_UARTE_Type *current = nrf_uarte_to_clean[i];
 
+        /* Ensured that RX is stopped */
+        if (nrfy_uarte_event_check(current, NRF_UARTE_EVENT_RXSTARTED)) {
+            nrfy_uarte_task_trigger(current, NRF_UARTE_TASK_STOPRX);
+            /* Wait up to 100ms for task to stop */
+            for (int i = 0; i < 1000; i++) {
```

Comment:
heard yesterday there is an deinit driver function in zephyr from @bjarki-andreasen not sure if it does the same as this or different but might be worth a look at

Reply:
The API is indeed implemented https://github.com/zephyrproject-rtos/zephyr/pull/93273 can't we just use it?